src/PSDMatrices.jl

```diff
@@ -1,13 +1,64 @@
+
 module PSDMatrices
 
-using LinearAlgebra
-import LinearAlgebra: cholesky, diag, det, logdet
+import Base: \, /, size, inv, copy, ==
+import LinearAlgebra: cholesky, det, logdet, qr, triu, diag, UpperTriangular
+
+struct PSDMatrix{T,Ltype} <: AbstractMatrix{T}
```

Comment:
I'm wondering if we should have an `abstract type AbstractPSDMatrix`?
See for instance: https://github.com/JuliaStats/PDMats.jl/blob/087d59fe70cc75667381bb5dac24f54ce59db26a/src/PDMats.jl#L36

Reply:
To be honest, I'm also not 100% sure, I just know that it's what everyone does. I also don't see much value personally right now, but I think that one aspect might also be that this way other people could implement their own subtypes of AbstractPSDMatrix, but still benefit from some of our code, if our functions operate on AbstractPSDMatrix and not PSDMatrix.

Reply:
> other people could implement their own subtypes of AbstractPSDMatrix
Is it true that at the moment, they could subtype PSDMatrix and have the same advantage?